frontend/packages/kubevirt-plugin/src/components/modals/cdrom-vm-modal/cdrom-modal.tsx

```diff
@@ -0,0 +1,219 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Form, Button, Tooltip, Alert } from '@patternfly/react-core';
+import {
+  FirehoseResult,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { isWindows } from 'kubevirt-web-ui-components';
+import { getName } from '@console/shared';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { PlusCircleIcon, OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import { K8sResourceKind, k8sPatch } from '@console/internal/module/k8s';
+import { getCDsPatch } from '../../../k8s/patches/vm/vm-cdrom-patches';
+import { getRemoveDiskPatches } from '../../../k8s/patches/vm/vm-disk-patches';
+import { getVMLikeModel } from '../../../selectors/vm';
+import {
+  getCDRoms,
+  getContainerImageByDisk,
+  getURLSourceByDisk,
+  getPVCSourceByDisk,
+  getStorageSizeByDisk,
+  getStorageClassNameByDisk,
+  isVMRunning,
+} from '../../../selectors/vm/selectors';
+import { VMKind } from '../../../types';
+import { StorageType } from '../../../constants/vm/storage';
+import { CDRomRow } from './cdrom-row';
+import { initialDisk, WINTOOLS_CONTAINER_NAMES } from './constants';
+import './_cdrom-modal.scss';
+
+export const CDRomModal = withHandlePromise((props: CDRomModalProps) => {
+  const {
+    vm,
+    handlePromise,
+    inProgress,
+    errorMessage,
+    persistentVolumeClaims,
+    storageClasses,
+    cancel,
+    close,
+  } = props;
+
+  const getCDsFromResource = (cds) =>
+    cds.map((disk) => {
+      const cd = {
+        ...initialDisk,
+        name: disk.name,
+        bootOrder: disk.bootOrder,
+        isInVM: true,
+      };
+      const container = getContainerImageByDisk(vm, cd.name);
+      if (container) {
+        if (_.includes(WINTOOLS_CONTAINER_NAMES, container))
+          return { ...cd, type: StorageType.WINTOOLS, windowsTools: container };
+        return { ...cd, type: StorageType.CONTAINER, container };
+      }
+
+      const url = getURLSourceByDisk(vm, cd.name);
```

Comment:
When I change the type to URL it doesn't show the storage class and size, maybe an issue with my environment?